src/FluentValidation/TestHelper/ValidatorTestExtensions.cs

```diff
@@ -48,6 +48,19 @@ public static class ValidationTestExtension {
 			return testValidationResult.ShouldHaveValidationErrorFor(expression);
 		}
 
+		public static IEnumerable<ValidationFailure> ShouldHaveValidationErrorFor<T>(this IValidator<T> validator, Expression<Func<T, T>> expression, string ruleSet)

```

Comment:
I think that this overload is confusing because it doesn't accept the model instance - it isn't clear that this method performs a check for *any* validation error. This behaviour can already be achieved through the advanced syntax, so let's remove this overload and just stick with the ones below.